catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/content/ContentProducerDataAccessObject.java

```diff
@@ -203,27 +218,18 @@ public void createContentItem(FileSystemPersistenceProvider fileIdMap, ContentEn
             if (deleteResponse != null) {
                 List<Metacard> deletedMetacards = deleteResponse.getDeletedMetacards();
 
-                for (Metacard delete : deletedMetacards) {
-                    fileIdMap.delete(ingestedFile.getAbsolutePath());
-                    LOGGER.debug("content item deleted with id = {}", delete.getId());
+                if (safeKey != null) {
+                    fileIdMap.delete(safeKey);
                 }
+                LOGGER.debug("content item deleted with id = {}",
+                        deletedMetacards.stream()
+                                .map(Metacard::getId));
             }
         }
     }
 
-    public void processHeaders(Map<String, Object> headers, StorageRequest storageRequest,
-            File ingestedFile) {
-        Map<String, Serializable> attributeOverrideHeaders = new HashMap<>((Map) headers.get(
-                Constants.ATTRIBUTE_OVERRIDES_KEY));
-        if (!attributeOverrideHeaders.isEmpty()) {
-            storageRequest.getProperties()
-                    .put(Constants.ATTRIBUTE_OVERRIDES_KEY,
-                            (Serializable) attributeOverrideHeaders);
-        }
-        if (headers.containsKey(Constants.STORE_REFERENCE_KEY)) {
-            storageRequest.getProperties()
-                    .put(Constants.STORE_REFERENCE_KEY, ingestedFile.getAbsolutePath());
-        }
+    protected HashMap<String, Serializable> getProperties(Map<String, Object> headers) {
+        return Maps.newHashMap(Maps.transformValues(headers, (v) -> (Serializable) v));
     }
 
     private void waitForAvailableSource(CatalogFramework catalogFramework)
```

Comment:
yeah, we will have to check out why this is still an issue. It has popped up on a few reviews. https://jira.sonarsource.com/browse/SONARJAVA-583 indicates it should be fixed.